xoa/data_source_hosts.go

```diff
@@ -68,7 +68,7 @@ func dataSourceHostsRead(d *schema.ResourceData, m interface{}) error {
 	}
 	err = d.Set("hosts", hosts)
 	if err != nil {
-		log.Print("[DEBUG] failed setting hosts")
+		log.Printf("[DEBUG] failed setting hosts: %s", err.Error())
```

Comment:
This error and its message will be shown in Terraform's output when the error is returned on line 72. Since this doesn't add any additional information, I think we should remove the log line entirely.